### PR TITLE
Stamp persona/model/settings into message metadata; drop the active-persona REST

### LINKED ISSUE #136: Deliver session state over chat metadata instead of REST polling
## Problem

The input-toolbar surface learns about persona and session state by polling REST: the frontend refetches `/active_persona` on every message update, plus one trailing refresh per burst, plus bounded polling while an agent session initializes. Each fetch re-serializes personas, controls, and usage even when nothing changed, while the chat already has a Yjs WebSocket syncing the document. Raised in review: https://github.com/jupyter-ai-contrib/jupyter-ai-acp-client/pull/133#discussion_r3554231358.

## Proposed solution

Store session-level state in chat metadata and let the existing Yjs channel deliver it: the frontend reads from the chat model it already syncs, and the REST polling goes away.

The groundwork half-exists: model, mode, and config selections already persist in chat metadata (`acp_models`, `acp_modes`, `acp_config_options`, keyed by persona id), so usage and controls become additional tenants of that channel, for example `ychat.metadata["persona-sessions"][persona.id]["usage"]` per the review sketch. Pairs with an upstream `BasePersona.update_usage()` in jupyter-ai-persona-manager, which would also give non-ACP personas a way to publish usage, and with lifting the UI components to the PersonaManager.

Design notes: metadata persists in the chat file while agent sessions reset on server restart, so stale usage needs clearing on session recreation; and `usage_update` can arrive mid-turn at chunk frequency, so metadata writes should be batched or throttled.

### PR DESCRIPTION
**Pull request by dlqqq**
## Motivation

This is the picker half of the v3.1 metadata-based routing work (persona-manager side: jupyter-ai-contrib/jupyter-ai-persona-manager#59). Routing now reads the target persona from each message's own metadata rather than server-side state, so the input toolbar needs to stamp the user's persona/model/settings selection onto the message it sends. Until now those controls fired REST calls to set an "active persona" on the server, which is exactly the shared, guess-at-state model we're moving away from.

## Summary

The persona/model/settings controls now write the current selection into the chat input's message metadata, so every message is self-describing and the PersonaManager routes on it. The selection lives entirely in the frontend and is seeded from the default persona the server advertises, so a fresh chat pre-selects a sensible persona without a round-trip. Switching persona or changing a control no longer sets any server-side "active persona".

## Technical details

- Requires the message-metadata accessors added in jupyter-chat (jupyterlab/jupyter-chat#465, released in 0.23.0a3), which this bumps to; the local type augmentation for those methods is dropped.

- The `/active_persona` endpoint becomes `/personas`, keyed on an optional persona id (whose controls to show), and the POST that set the active persona is gone. Naming a persona that isn't installed returns no controls rather than erroring, so the selector still renders. `/control` now takes the persona id explicitly. This REST path stays only because control changes must reach the agent's live ACP session, which metadata doesn't drive yet.

- Rebased on #133; the usage chip and this selection metadata coexist.

## Testing

Backend route tests and frontend unit tests for the metadata builder and persona resolution; built and exercised the picker locally against a running server.

### CODE REVIEW

**Comment by andrii-i** [comment]
* Fixes #136

File: src/persona-controls.tsx [hunk 1]
```diff
@@ -687,18 +741,32 @@ export function AcpPersonaControls(
...
+  // present in this chat, fall back to the sole persona (if there's exactly
+  // one) or to no one, so we never point at a persona the chat doesn't have.
+  useEffect(() => {
+    if (!personas.length) {
+      return;
+    }
+    if (selectedId && personas.some(p => p.id === selectedId)) {
+      return;
+    }
+    setSelectedId(personas.length === 1 ? personas[0].id : null);
```
Comment (andrii-i) [inline src/persona-controls.tsx:754]: Should this fallback be gated on the user not having picked yet, e.g. a `userPicked` ref, so that picking `No one` in a chat with exactly one persona doesn't get overwritten (seems like it re-fires on every refresh since `personas` is a new array each time) and `No one` stays selectable there? Multi-persona chats are unaffected, the same line already resolves to null for them.

File: jupyter_ai_acp_client/routes.py [hunk 2]
```diff
@@ -269,31 +269,47 @@ def build_usage(persona: BaseAcpPersona) -> AcpUsage:
...
+        else the chat's default persona. Returns None if neither resolves.
+
+        A `persona_id` naming a persona that isn't installed here (e.g. a default
+        advertised by the server but not present in this environment) is not an
+        error: it just means "no controls to show". We must still return the
+        persona list so the selector can render, so this never raises."""
+        persona_id = self.get_argument("persona_id", None)
+        if persona_id:
+            return persona_manager.personas.get(persona_id)
+        return persona_manager.default_persona
```
Comment (andrii-i) [inline jupyter_ai_acp_client/routes.py:306]: Should no `persona_id` return no controls now that the initial selection is seeded from PageConfig, so that the `No one` selection (seems to be the only case where the frontend omits `persona_id`) doesn't end up showing the default persona's controls?

File: src/persona-controls.tsx [hunk 3]
```diff
@@ -711,39 +779,59 @@ export function AcpPersonaControls(
...
 
-  const activeAvatar =
-    personas.find(p => p.id === activeId)?.avatar_url ?? null;
+  const selectedPersona = personas.find(p => p.id === selectedId) ?? null;
+  const activeName = selectedPersona?.name ?? null;
+  const activeAvatar = selectedPersona?.avatar_url ?? null;
   const personaLabel = activeName ?? NO_ONE_LABEL;
 
-  const handlePersona = async (personaId: string | null) => {
+  const handlePersona = (personaId: string | null) => {
```
Comment (andrii-i) [inline src/persona-controls.tsx:805]: Should this also `setControls([])` to prevent a switch-then-send from potentially stamping the previous persona's `model`/`settings` under the new `to_persona`?

File: jupyter_ai_acp_client/routes.py [hunk 4]
```diff
@@ -55,7 +55,7 @@ def get(self, persona_mention_name: str = ""):
                 # raise HTTP error: persona not found
                 raise tornado.web.HTTPError(404, f"Persona not found: @{persona_mention_name}")
         else:
-            persona = persona_manager.active_persona
+            persona = persona_manager.default_persona
```
Comment (andrii-i) [inline jupyter_ai_acp_client/routes.py:58]: Should `listCommandCompletions` pass `inputModel.getMetadata().to_persona`, with this endpoint accepting a persona id, so that `/` completions follow the picker selection? They seem to come from `default_persona` regardless of the selection now (locally I get no ACP commands for any selected persona, since my default doesn't resolve), and the upcoming mention removal would make this the only path.

**Review by andrii-i — APPROVED** [review]
Thanks, this is a clear improvement vs shared server-side active-persona state. Tested and it works well other than the edge cases below. I can address them in my follow-up PR, or feel free to address some of them before merging.

File: src/persona-controls.tsx (hunk 1 above)
Comment (dlqqq) [reply src/persona-controls.tsx:754]: Agree we need to think through this a bit more. For now, we expect that if it is set, that persona must always be available. This safety check was added by AI automatically as a stopgap.

File: src/persona-controls.tsx (hunk 3 above)
Comment (dlqqq) [reply src/persona-controls.tsx:805]: Great catch! Yes, we definitely need to handle this edge case. This might get fixed automatically when we implement https://github.com/jupyter-ai-contrib/jupyter-ai-acp-client/pull/135 however. I'll open a follow-up issue to make sure we at least test this edge case.

Comment (dlqqq) [reply src/persona-controls.tsx:805]: Opened https://github.com/jupyter-ai-contrib/jupyter-ai-persona-manager/issues/62 as follow up.

File: jupyter_ai_acp_client/routes.py (hunk 4 above)
Comment (dlqqq) [reply jupyter_ai_acp_client/routes.py:58]: The previous slash command logic relied on knowing which persona was the active persona, and now we've removed that concept entirely. Best to address this in another PR I think.

File: jupyter_ai_acp_client/routes.py (hunk 2 above)
Comment (dlqqq) [reply jupyter_ai_acp_client/routes.py:306]: See prev comment regarding similar callout.
